src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java

```diff
@@ -73,6 +81,42 @@ public boolean isOperationsUser(RestRequest request) {
         }
         return allowed;
     }
+    
+    @SuppressWarnings("rawtypes")
+    public void assertUser(RestRequest request) throws Exception {
+        final String user = getUser(request);
+        final String token = getBearerToken(request);
+        ConfigBuilder builder = new ConfigBuilder().withOauthToken(token);
+        try (DefaultOpenShiftClient osClient = new DefaultOpenShiftClient(builder.build())) {
+            LOGGER.debug("Verifying user {} matches the given token.", user);
+            Request okRequest = new Request.Builder()
+                .addHeader(AUTHORIZATION_HEADER, "Bearer " + token)
+                .url(osClient.getMasterUrl() + "oapi/v1/users/~") 
+                .build();
+            String username = null;
+            try (Response response = osClient.getHttpClient().newCall(okRequest).execute()){
+                if(response.code() != RestStatus.OK.getStatus()) {
+                    throw new ElasticsearchSecurityException("The user is unauthorized", RestStatus.UNAUTHORIZED);
```

Comment:
always ok to hide the response code here with unauthorized?  Maybe add a debug for the real response code just in case.

Reply:
I wanted to throw an exception with the returned code from the other call but the only class that take s a code was "SecurityException"

Reply:
Removed message and modified to have debug log message with code